In [1]:
import pandas as pd
from pandas import DataFrame
import numpy as np

In [2]:
# read in the csv 
df = pd.read_csv('purchase_data.csv')

In [3]:
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [4]:
# total number of players
total_players = len(df['SN'])
DataFrame({'Total Players':[total_players]})

Total Players
0            780

In [5]:
# total number of unique items
unique_items = df['Item Name'].nunique()
player_count = DataFrame({'Unique Items':[unique_items]})
player_count

Unique Items
0           179

In [6]:
# average purchase price
avg_price = round(df['Price'].mean(),2)
avg_price = '${:,.2f}'.format(avg_price)
avg_price

'$3.05'

In [7]:
# total number of purchases
num_of_purchases = df['Purchase ID'].nunique()
num_of_purchases

780

In [8]:
# total revenue
total_rev = df['Price'].sum()
total_rev = '${:,.2f}'.format(total_rev)
total_rev

'$2,379.77'

In [9]:
# purchasing analysis
purchasing_analysis= DataFrame({'Number of Unique Items':[unique_items],'Average Price':[avg_price],
                               'Number of Purchases':[num_of_purchases],'Total Revenue':[total_rev]})
purchasing_analysis

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

In [10]:
gender = df['Gender'].value_counts()
percent_of_players = (gender/total_players) 
#percent_of_players
percent_of_players = percent_of_players.apply('{:1.1%}'.format)




In [11]:
gender_demographics = DataFrame({'Percentage of Players':percent_of_players,'Total Count': gender})
gender_demographics

Percentage of Players  Total Count
Male                                  83.6%          652
Female                                14.5%          113
Other / Non-Disclosed                  1.9%           15

In [12]:
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [13]:
# purchasing analysis (Gender)
purchase_count = df.groupby('Gender')
purchase_count = purchase_count['Purchase ID'].count()
purchase_count = purchase_count.apply('{:1.1%}'.format)

In [14]:
purchase_price = df.groupby('Gender')
purchase_price = purchase_price['Price'].mean()
purchase_price = purchase_price.apply('${:,.2f}'.format)
purchase_price

Gender
Female                   $3.20
Male                     $3.02
Other / Non-Disclosed    $3.35
Name: Price, dtype: object

In [15]:
purchase_total = df.groupby('Gender')
purchase_total = purchase_total['Price'].sum()
purchase_total = purchase_total.apply('${:,.2f}'.format)
purchase_total

Gender
Female                     $361.94
Male                     $1,967.64
Other / Non-Disclosed       $50.19
Name: Price, dtype: object

In [16]:
purchasing_analysis = DataFrame({'Purchase Count': gender, 'Average Purchase Price':purchase_price,
                                'Total Purchase Value': purchase_total,'Avg Purchase Total per Person':purchase_price})
purchasing_analysis

Purchase Count Average Purchase Price  \
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Purchase Total per Person  
Female                             $361.94                         $3.20  
Male                             $1,967.64                         $3.02  
Other / Non-Disclosed               $50.19                         $3.35

In [17]:
# age_demographics
age_bins = [0, 9.90, 14.90, 19.90, 24.90, 29.90, 34.90, 39.90, 99999]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

In [18]:
df['Age Groups'] = pd.cut(df['Age'], age_bins, labels = group_names)


In [19]:
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Groups  
0  Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                          Frenzied Scimitar   1.56        40+  
2                               Final Critic   4.88      20-24  
3                                Blindscythe   3.27      20-24  
4                                       Fury   1.44      20-24

In [20]:
age_groups = df['Age Groups'].value_counts()
age_groups

20-24    365
15-19    136
25-29    101
30-34     73
35-39     41
10-14     28
<10       23
40+       13
Name: Age Groups, dtype: int64

In [21]:
age_percents = age_groups / total_players
age_percents = age_percents.apply('{:1.2%}'.format)
age_percents

20-24    46.79%
15-19    17.44%
25-29    12.95%
30-34     9.36%
35-39     5.26%
10-14     3.59%
<10       2.95%
40+       1.67%
Name: Age Groups, dtype: object

In [22]:
age_demographics = DataFrame({'Percentage of Players': age_percents,'Total Count': age_groups})
age_demographics

Percentage of Players  Total Count
20-24                46.79%          365
15-19                17.44%          136
25-29                12.95%          101
30-34                 9.36%           73
35-39                 5.26%           41
10-14                 3.59%           28
<10                   2.95%           23
40+                   1.67%           13

In [23]:
# purchasing analysis (Age)
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Groups  
0  Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                          Frenzied Scimitar   1.56        40+  
2                               Final Critic   4.88      20-24  
3                                Blindscythe   3.27      20-24  
4                                       Fury   1.44      20-24

In [24]:
# top spender analysis 
top_spender_count = df['SN'].value_counts()


In [25]:
average_spend_price = df.groupby('SN')
average_spend_price = average_spend_price['Price'].mean()
average_spend_price = average_spend_price.apply('${:,.2f}'.format)


In [26]:
total_spend_price = df.groupby('SN')
total_spend_price = total_spend_price['Price'].sum()
total_spend_price = total_spend_price.apply('${:,.2f}'.format)

In [27]:
top_spenders = DataFrame({'Purchase Count':top_spender_count,'Average Purchase Price':average_spend_price,
                        'Total Purchase Value':total_spend_price})
top_spenders.sort_values(by=['Purchase Count','Average Purchase Price','Total Purchase Value'],ascending=False,inplace=True)
top_spenders.head()

Purchase Count Average Purchase Price Total Purchase Value
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Iral74                    4                  $3.40               $13.62
Chamjask73                3                  $4.61               $13.83
Iskadarya95               3                  $4.37               $13.10

In [28]:
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Groups  
0  Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                          Frenzied Scimitar   1.56        40+  
2                               Final Critic   4.88      20-24  
3                                Blindscythe   3.27      20-24  
4                                       Fury   1.44      20-24

In [29]:
# most popular items
item_count = df.groupby(['Item ID'])
item_count = DataFrame(item_count['Price'].count())


In [30]:
item_price = df.groupby(['Item ID'])
item_price = DataFrame(item_price['Price'].sum())
average_purch_price = df.groupby(['Item ID'])
average_purch_price = DataFrame(average_purch_price['Price'].mean())


In [31]:
mergedata = pd.merge(item_count,item_price,how='inner',on='Item ID')
mergedata = mergedata.rename(columns={'Price_x':'Purchase Count','Price_y':'Total Purchase Value'})
mergedata = pd.merge(mergedata,average_purch_price,how='inner',on='Item ID')
mergedata = mergedata.rename(columns={'Price':'Average Purchase Price'})
mergedata.head()

Purchase Count  Total Purchase Value  Average Purchase Price
Item ID                                                              
0                     4                  5.12                    1.28
1                     3                  9.78                    3.26
2                     6                 14.88                    2.48
3                     6                 14.94                    2.49
4                     5                  8.50                    1.70

 Heroes Of Pymoli Data Analysis

Of the 780 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female players (~15%).
Despite women making up a smaller portion of total players. They on average spend slightly more than men for in-app purchases. 

In [32]:
test = item_count[item_count['Price'] > 5.00].count()
print(test)

Price    40
dtype: int64
